[data.search] Remove deprecated courier:batchSearches advanced setting

lukasolson opened the pull request:
## Summary

Resolves https://github.com/elastic/kibana/issues/55140.

Removes the `courier:batchSearches` advanced setting and its usage, as it has been deprecated since 7.6 (https://github.com/elastic/kibana/pull/54539).

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/master/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/master/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Release note

Removes the deprecated `courier:batchSearches` advanced setting which allowed using the Elasticsearch `_search`API instead of `_async_search`.

elasticmachine commented:
Pinging @elastic/kibana-app-services (Team:AppServices)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/)
* Commit: da702fe1381ff419ce4b487a7c6a91734b30cd6d
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Doc API Changes](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/execution/node/407/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/","number":"147147","commit":"da702fe1381ff419ce4b487a7c6a91734b30cd6d"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147157/)
* Commit: 35096bfcf2795309576bed2da89281a2b48ab03d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109350/35096bfcf2795309576bed2da89281a2b48ab03d)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147157/execution/node/190/log)

### History
* :broken_heart: [Build #147147](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/) failed da702fe1381ff419ce4b487a7c6a91734b30cd6d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/","number":"147147","commit":"da702fe1381ff419ce4b487a7c6a91734b30cd6d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147157/","number":"147157","commit":"35096bfcf2795309576bed2da89281a2b48ab03d"}]}
PIPELINE-->

mshustov reviewed (approved):
Ok for the core changes

Dosant commented:
Do we leave it as 8.0 change or do we also back port this to 7.x?

lukasolson commented:
> Do we leave it as 8.0 change or do we also back port this to 7.x?

Thanks for catching this, I added the 7.x and auto-backport labels out of habit. I've removed them now.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147433/)
* Commit: e160be3e37b073ba8882ebb707ef47bcc0fffd18
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109350/e160be3e37b073ba8882ebb707ef47bcc0fffd18)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147433/execution/node/713/log)
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147433/execution/node/792/log)

### History
* :broken_heart: [Build #147157](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147157/) failed 35096bfcf2795309576bed2da89281a2b48ab03d
* :broken_heart: [Build #147147](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/) failed da702fe1381ff419ce4b487a7c6a91734b30cd6d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/","number":"147147","commit":"da702fe1381ff419ce4b487a7c6a91734b30cd6d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147157/","number":"147157","commit":"35096bfcf2795309576bed2da89281a2b48ab03d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147433/","number":"147433","commit":"e160be3e37b073ba8882ebb707ef47bcc0fffd18"}]}
PIPELINE-->

Dosant commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147650/)
* Commit: 3d725a3002f98d7d6d57ec57833db7c87f5e9ffd
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109350/3d725a3002f98d7d6d57ec57833db7c87f5e9ffd)
* [Documentation Changes](https://kibana_109350.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/92c6787af115edfcd36637a5f2e932310c6f270d) | [after](https://github.com/elastic/kibana/commit/3d725a3002f98d7d6d57ec57833db7c87f5e9ffd) | diff |
| --- | --- | --- | --- |
| `data` | 549 | 547 | -2 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/92c6787af115edfcd36637a5f2e932310c6f270d) | [after](https://github.com/elastic/kibana/commit/3d725a3002f98d7d6d57ec57833db7c87f5e9ffd) | diff |
| --- | --- | --- | --- |
| `data` | 790.1KB | 788.9KB | -1.2KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/92c6787af115edfcd36637a5f2e932310c6f270d) | [after](https://github.com/elastic/kibana/commit/3d725a3002f98d7d6d57ec57833db7c87f5e9ffd) | diff |
| --- | --- | --- | --- |
| `data` | 3502 | 3498 | -4 |

#### API count missing comments
| id | [before](https://github.com/elastic/kibana/commit/92c6787af115edfcd36637a5f2e932310c6f270d) | [after](https://github.com/elastic/kibana/commit/3d725a3002f98d7d6d57ec57833db7c87f5e9ffd) | diff |
| --- | --- | --- | --- |
| `data` | 2985 | 2981 | -4 |


</details>

### History
* :broken_heart: [Build #147433](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147433/) failed e160be3e37b073ba8882ebb707ef47bcc0fffd18
* :broken_heart: [Build #147157](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147157/) failed 35096bfcf2795309576bed2da89281a2b48ab03d
* :broken_heart: [Build #147147](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/) failed da702fe1381ff419ce4b487a7c6a91734b30cd6d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147147/","number":"147147","commit":"da702fe1381ff419ce4b487a7c6a91734b30cd6d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147157/","number":"147157","commit":"35096bfcf2795309576bed2da89281a2b48ab03d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147433/","number":"147433","commit":"e160be3e37b073ba8882ebb707ef47bcc0fffd18"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147650/","number":"147650","commit":"3d725a3002f98d7d6d57ec57833db7c87f5e9ffd"}]}
PIPELINE-->

Dosant reviewed (approved):
LGTM since ci is happy. didn't test

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 109350` or prevent reminders by adding the `backport:skip` label.
